src/main/java/com/based/basedsurvey/controller/FillSurveyController.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import java.util.List;
+
+@Controller
+public class FillSurveyController {
+
+    SurveyRepository surveyRepository;
+
+    @Autowired
+    public FillSurveyController(SurveyRepository surveyRepository) {
+        this.surveyRepository = surveyRepository;
+    }
+
+    @GetMapping("/survey/{surveyID}/answer")
+    public String getAnswer(@PathVariable Long surveyID, Model model) {
+        // if survey does not exist
+        if (!surveyRepository.existsById(surveyID)) {
+            model.addAttribute("issue", "Survey does not exist");
+            return "SurveyFillIssue";
+        }
+
+        model.addAttribute("surveyID", surveyID);
+        Survey s = this.surveyRepository.findSurveyById(surveyID);
+        model.addAttribute("surveyName", s.getName());
+
+        // if survey is closed
+        if (!s.isOpen()) {
+            model.addAttribute("issue", "Survey is closed");
+            return "SurveyFillIssue";
+        }
+
+        // if survey has no questions
+        if (s.getQuestions().size() == 0) {
+            model.addAttribute("issue", "Survey has no questions");
+            return "SurveyFillIssue";
+        }
+
+        // formInputs will contain a dynamically generated string with all the html form inputs
+        String formInputs = "";
+
+        // for each question in survey
+        for (Question q : s.getQuestions()) {
+
+            // case 1, the question is a multiple choice question
+            if (q instanceof MultiplechoiceQuestion) {
+                formInputs += getMultipleChoiceInputs((MultiplechoiceQuestion) q);
+
+                // case 2, the question is an open answer question
+            } else if (q instanceof OpenAnswerQuestion) {
+                formInputs += getOpenAnswerInput((OpenAnswerQuestion) q);
+
+                // case 3, the question is a range question
+            } else if (q instanceof RangeQuestion) {
+                formInputs += getRangeInput((RangeQuestion) q);
+            }
+        }
+
+        model.addAttribute("questions", formInputs);
+        return "Answer";
+    }
+
+    // side note, html forms can't send patch requests and I used post instead
+    @PostMapping("/survey/{surveyID}/answer")
+    public String postAnswer(@PathVariable Long surveyID, @RequestParam List<String> values, Model model) {
+        // if survey does not exist
+        if (!surveyRepository.existsById(surveyID)) {
+            model.addAttribute("issue", "Survey does not exist");
+            return "SurveyFillIssue";
+        }
+
+        Survey s = this.surveyRepository.findSurveyById(surveyID);
+
+        // for each question in survey add the response
+        for (int index = 0; index < s.getQuestions().size(); index += 1) {
+            Question q = s.getQuestions().get(index);
+            addResponse(q, values.get(index));
+        }
+
+        surveyRepository.save(s);
```

Comment:
this works, multiple responses can be stored, nice

Reply:
based even